app/src/org/odk/collect/android/logic/ArchivedFormRemoteRestore.java

```diff
@@ -0,0 +1,131 @@
+package org.odk.collect.android.logic;
+
+import android.widget.Toast;
+
+import org.commcare.android.database.user.models.User;
+import org.commcare.android.framework.CommCareActivity;
+import org.commcare.android.tasks.DataPullTask;
+import org.commcare.android.tasks.FormRecordCleanupTask;
+import org.commcare.android.tasks.templates.CommCareTaskConnector;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.activities.FormRecordListActivity;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Only used for developer debugging.
+ *
+ * Load saved form instances manually from a xml payload.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class ArchivedFormRemoteRestore<A extends CommCareActivity> {
```

Comment:
I over generalized here because I thought I was going to use this task from within a test, as well as where it is being used in the app code. I figured I'd leave it in this state in case we do ever need it in a test (probably super unlikely...). I can remove the generics if you think that is the best way to leave things.

I don't want to mix it with `ArchivedFormManagement` because that code is actually used in production, whereas this is a debug feature that manually has to be enabled by the dev (ugly, but useful at times)